### PR TITLE
add patchMetadata for vault

### PR DESCRIPTION
**Pull request by kengou**
added a func to patch Metadata

### CODE REVIEW

File: pkg/vault/client.go [hunk 1]
```diff
@@ -136,7 +137,30 @@ func (c *Client) UpdateCertificate(data CertificateData) error {
 
 	if needsWrite {
 		_, err := c.client.Logical().Write(fullSecretPath, map[string]interface{}{"data": payload})
+		if err != nil {
+			return fmt.Errorf("while wrinting payload to vault: %w", err)
```
Comment (auhlig) [inline pkg/vault/client.go:141]:
```suggestion
			return fmt.Errorf("while writing payload to vault: %w", err)
```

**Review by auhlig — APPROVED** [review]

**Comment by databus23** [comment]
We don't put in any support group or ownership afaics. I guess establishing ownership is hard, but maybe we can copy over the support-group from the source crd?

File: pkg/vault/client.go [hunk 2]
```diff
@@ -136,7 +137,30 @@ func (c *Client) UpdateCertificate(data CertificateData) error {
 
 	if needsWrite {
```
Comment (Nuckal777) [inline pkg/vault/client.go:138]: Currently, the metadata is only updated when the certificate changes. For certificates with long validity that may be to infrequent. I suggest to compare the expected metadata with the current metadata (which should be returned by the secret read) and issue a patchMetadata based on that.

**Comment by Nuckal777** [comment]
`PatchMetadata` should use a different API endpoint than writing/reading secrets. It may be necessary to adjust the vault policies used by git-cert-shim.

**Comment by kengou** [comment]
> We don't put in any support group or ownership afaics. I guess establishing ownership is hard, but maybe we can copy over the support-group from the source crd?

I was not able to find anything related in the `Certificate` nor in the `CertificateRequest` CRD. Any other guess where can we find a hint for the ownership ?

File: pkg/vault/client.go [hunk 3]
```diff
@@ -136,7 +137,30 @@ func (c *Client) UpdateCertificate(data CertificateData) error {
...
+	date := fmt.Sprintf("%d-%02d-%02d", t.Year(), t.Month(), t.Day())
+	customMetadata := map[string]interface{}{
+		"accessed_resource":       c.client.Address(),
+		"application_criticality": "high",
+		"expiry_date":             date,
+		"review_date":             date,
+		"is_privileged":           "false",
+		"is_single_factor":        "false",
+		"username":                "UNLINKED",
+	}
```
Comment (majewsky) [inline pkg/vault/client.go:160]: This is fine within the scope of this PR, but in the midterm, I would like to have this not overwrite more specific values set by the owner of the cert.

**Review by databus23 — APPROVED** [review]

**Pull request merged**